### PR TITLE
remove extension registry UI and related GraphQL API

### PR DESCRIPTION
**Pull request by sqs**
Removes the extension registry UI and the GraphQL API that is necessary for the registry extension UI.

Any extensions that already exist locally or remotely can still be used with the enableLegacyExtensions site config flag and by enabling the extensions in your user settings.

This change is an intermediate step toward removing the extension registry (except the endpoint that's needed to support pre-4.0 Sourcegraph instances).

I think this PR has a 99% chance of going into the 4.4 release (2023-01-20) and not needing to be reverted. Subsequent PRs that actually make it impossible to use local or custom legacy extensions are still pending customer communication and confirmation. I hope we can merge other future PRs there as well, but there is a greater chance we'd need to revert  them before 4.4. By making this PR first, it means less of a revert and more progress regardless of what we learn from customers.

See [more internal context](https://sourcegraph.slack.com/archives/C03P1V3J92P/p1671668923668759) and the [tracking gdoc](https://docs.google.com/document/d/10vtoe-kpNvVZ8Etrx34bSCoTaCCHxX8o3ncCmuErPZo/edit#).

## Test plan

1. Add `"extensions": {"sourcegraph/github-extras": true}` to your user settings.
2. Visit https://sourcegraph.test:3443/github.com/sourcegraph/create/-/blob/prettier.config.js#tab=github.pullRequests.
3. Expected: a panel tab with PRs.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| -0.48% (-13.70 kb) 🔽 | -1.33% (-197.25 kb) 🔽 | -1.53% (-183.55 kb) 🔽 | -2.11% (-16) 🔽 |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/sqs/rm-extension-api-ui/compare-report.html;tab=live_object) for a full comparison between the commits bd25c86 and 0e3ebd5 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/web/src/extensions/extension/ExtensionArea.tsx [hunk 1]
```diff
@@ -1,275 +0,0 @@
...
-import AlertCircleIcon from 'mdi-react/AlertCircleIcon'
-import MapSearchIcon from 'mdi-react/MapSearchIcon'
-import { Route, RouteComponentProps, Switch } from 'react-router'
-import { combineLatest, merge, Observable, of, Subject, Subscription } from 'rxjs'
-import { catchError, distinctUntilChanged, map, mapTo, startWith, switchMap } from 'rxjs/operators'
-
-import { createAggregateError, ErrorLike, isErrorLike, asError, logger } from '@sourcegraph/common'
-import { gql } from '@sourcegraph/http-client'
-import {
-    ConfiguredRegistryExtension,
```
Comment (valerybugakov) [inline client/web/src/extensions/extension/ExtensionArea.tsx:12]: Can we remove this interface too?

File: client/web/src/extensions/ExtensionRegistry.tsx [hunk 2]
```diff
@@ -1,466 +0,0 @@
...
-      })
-    | ErrorLike
-
-export type ExtensionsEnablement = 'all' | 'enabled' | 'disabled'
-
-export type ExtensionCategoryOrAll = ExtensionCategory | 'All'
-
-const extensionRegistryQuery = gql`
-    query RegistryExtensions($query: String, $prioritizeExtensionIDs: [String!]!, $getFeatured: Boolean!) {
-        extensionRegistry {
```
Comment (valerybugakov) [inline client/web/src/extensions/ExtensionRegistry.tsx:64]: Is it safe to delete [getExtensionManifests](https://sourcegraph.com/search?q=context%3Aglobal+repo%3A%5Egithub%5C.com%2Fsourcegraph%2Fsourcegraph%24+rev%3Asqs%2Frm-extension-api-secondary-uis+getExtensionManifests&patternType=standard&sm=1&groupBy=path) from the code-intel API? It's not directly related to the deleted file, but it came up in my search queries.

**Review by valerybugakov — APPROVED** [review]
Such a pleasure to see a diff like this! 🤩

**Pull request merged**